app/components/Poll/index.js

```diff
@@ -0,0 +1,152 @@
+// @flow
+
+import React from 'react';
+import Button from 'app/components/Button';
+import styles from './Poll.css';
+import type { PollEntity } from 'app/reducers/polls';
+import { Link } from 'react-router';
+import Icon from 'app/components/Icon';
+
+type Props = {
+  poll: PollEntity,
+  handleVote: (pollId: number, optionId: number) => Promise<*>,
+  backgroundLight?: boolean,
+  truncate?: number,
+  details?: boolean
+};
+
+type State = {
+  truncateOptions: boolean,
+  optionsToShow: Array<Object>,
+  expanded: boolean
+};
+
+class Poll extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+    const { options } = props.poll;
+    if (props.truncate && options.length > props.truncate) {
+      this.state = {
+        truncateOptions: true,
+        optionsToShow: options.slice(0, 3),
+        expanded: false
+      };
+    } else if (!props.truncate) {
+      this.state = {
+        truncateOptions: false,
+        optionsToShow: options,
+        expanded: true
+      };
+    } else {
```

Comment:
Isn't these two last branches equal in terms of outcome?

Reply:
Looks like it yeah.